File: Frontend/src/types/car.ts
export interface CreateCarRequest {
  make: string;
  model: string;
  licensePlate: string;
  year: number;
}

export interface UpdateCarRequest {
  make: string;
  model: string;
  licensePlate: string;
  year: number;
}

export interface CarResponse {
  id: string;
  make: string;
  model: string;
  licensePlate: string;
  year: number;
  isAvailable: boolean;
  createdAt: string;
}

File: Api/Models/Car.cs
namespace API.Models;

public class Car
{
    public Guid Id { get; set; }
    public required string Make { get; set; }
    public required string Model { get; set; }
    public required string LicensePlate { get; set; }
    public int Year { get; set; }
    public bool IsAvailable { get; set; } = true;
    public DateTimeOffset CreatedAt { get; set; }

    public ICollection<Rental> Rentals { get; set; } = [];
}
